### PR TITLE
Add fixes for updating gems for Resque

### PR DESCRIPTION
**Pull request by dbussink**

### CODE REVIEW

File: test/resque_test.rb [hunk 1]
```diff
@@ -21,6 +21,16 @@
     assert_equal 'namespace', Resque.redis.namespace
   end
 
+  test "can a redis URL" do
```
Comment (zerowidth) [inline test/resque_test.rb]:
```suggestion
  test "can set the redis connection to a redis URL" do
```

File: test/resque_test.rb [hunk 2]
```diff
@@ -21,6 +21,16 @@
...
+    Resque.redis = 'redis://localhost:9736'
+    assert_equal 'redis://localhost:9736', Resque.redis._client.options[:url]
+  end
+
+  test "can a unix socket URL" do
```
Comment (zerowidth) [inline test/resque_test.rb]:
```suggestion
  test "can set the redis connection a unix socket URL" do
```

**Review by zerowidth — APPROVED** [review]

File: lib/resque.rb [hunk 3]
```diff
@@ -181,11 +180,35 @@ def push(queue, item)
...
-    queue && [queue.sub("queue:", ""), decode(payload)]
+    begin
+      queue, payload = redis.blpop(*queue_names, timeout)
+      queue && [queue.sub("queue:", ""), decode(payload)]
+    rescue Redis::TimeoutError => e
+      # This exception happens when Redis goes
+      # away or we failover through a proxy layer.
+      if reconnect(retries)
+        # Wait for a random time between 0 and 1 second to prevent thundering reconnect herd
+        sleep rand
```
Comment (zerowidth) [inline lib/resque.rb]: Is this second sleep (the first being within `#reconnect`) necessary? If it's successfully reconnected, should we wait any longer to resume operation?

Comment (dbussink) [reply lib/resque.rb]: Good catch, this was intended to be above the `reconnect` but I had been shuffling around the code a few times. Moving it.

**Pull request merged**